### ISSUE TITLE
Weird search behavior

### ISSUE THREAD

**Issue by okrick** [DistributedProofreaders/guiguts-py#372]
I'm getting a weird result on the following regex search and replace:
Search  `([ \n])[“”]([\n ])`
Replace `\1„\2` 

The search always skips the second line quote (José Rodriguez) in the following but finds the quote in line three (Donna Maria).    It seems to find an end of line spaced quote more often than not but there are other missed places.  If I narrow the search term to ` ”\n` then it apparently finds all of the lines with issues.  The last three characters in both lines are "Space " "Right Double Quotation Mark" "Line Feed" per the GG2 Character Name Display.

```
/*
João Baptista Lacenda                      6,000 acres.
José Rodriguez                             1,500   ”
Donna Maria (Mai de Major Joaquim Roberto) 1,200   ”
The first can be bought for about               $5,000.
The second can be bought for about              $3,000.
The last can be bought for about                $2,500.
*/
```

Is this a 'feature' of the way Python treats searches?  If so then how should the search term be rephrased?
[brazil-src.zip](https://github.com/user-attachments/files/16530593/brazil-src.zip)

**Comment by okrick** [comment]
Changed the search term to ` [“”]\n`  then tried to Replace (Replace, R & S, and Replace All have the same error) the two terms skipped by the previous search.  The replacement term was `„`.  I got the following error message:
![Capture](https://github.com/user-attachments/assets/2d4ac917-2327-42c4-9124-972a93cd12fb)

**Comment by windymilla** [comment]
Definitely a bug - varying the regex slightly and using different start positions gives different results.
The regex searching within the text window does concern me - it seems to fail to find some things it should.
See issue #266 for another example, some links, and possible solution(s)

**Comment by windymilla** [comment]
> Is this a 'feature' of the way Python treats searches? If so then how should the search term be rephrased?

@okrick - I think the answer is "Yes, at least for now". If you recall, GG1 has to do something special when a regex contains a `\n` character. GG2 may need something similar under some circumstances - I'll add that to the existing issue we have relating to regex improvements.

In this particular case, you aren't actually interested in the newline character itself, so you can use the following regex, which uses `^` to mean "start of line" and `$` to mean "end of line":
`( |^)[“”]( |$)`
I've tested it out on that example file, and also trying ditto mark at the start of a line, and it seems to work OK.

Hope that helps - if it's fine, then I'd like this issue to be closed, and we'll leave it to issue #144 to deal with the problems.
